python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py

```diff
@@ -101,12 +106,23 @@ def _execUpdateAction(self):
                 else:
                     action.callback_Failure()
 
+            self._currently_running_command = False
+
 # Create the thread pool
 
 
 def _makeThreadPool(threadPoolSize=THREAD_POOL_SIZE, daemonic=True):
+    global ThreadPool
+    if ThreadPool and len(ThreadPool) != 0:
+        #from Ganga.Core.exceptions import GangaException
+        #raise GangaException("Cannot doubbly init the ThreadPool! ThreadPool already populated with threads")
+        log.debug("Found a thread pool already in existance, wiping it and startig again!")
+        for i in ThreadPool:
+            del i
+        del ThreadPool[:]
+        ThreadPool = []
```

Comment:
It is indeed these changes which is now causing the test suite to run a lot slower. If it's fixing things then I agree that it's not itself a problem as 'correct' beats 'fast'. Two things though. Firstly, I believe that:
```python
for i in ThreadPool:
    del i
del ThreadPool[:]
ThreadPool = []
```
could all just be replaced with
```python
del ThreadPool[:]
```

The second and most important thing to realise here though is that just because the threads have been 'deleted' (had this reference removed) that doesn't mean that they will stop running. If one if them was doing something which is hanging (maybe talking to DIRAC is taking many minutes) then running this code will not halt that process, it will continue to run in the background. This could cause situations where we have two DIRAC monitoring threads running simultaneously.

In Python there is essentially no way to kill a thread dead if it's busy. The only safe thing to do in reality is to wait for the threads to to finish.